quote-book/src/synchronized_quote_book.rs

```diff
@@ -90,6 +163,156 @@ where
     }
 }
 
+/// State that we want to expose from the db poll thread
+#[derive(Debug, Default)]
+pub struct DbPollSharedState {
+    /// The highest block index for which we can guarantee we have loaded all
+    /// available data.
+    pub highest_processed_block_index: u64,
+}
+struct DbFetcherThread<DB: Ledger, Q: QuoteBook> {
+    db: DB,
+    quotebook: Q,
+    /// Message bus sender.
+    msg_bus_tx: Sender<Msg>,
+    next_block_index: u64,
+    shared_state: Arc<Mutex<DbPollSharedState>>,
+    stop_requested: Arc<AtomicBool>,
+    logger: Logger,
+}
+
+/// Background worker thread implementation that takes care of periodically
+/// polling data out of the database. Add join handle
+impl<DB: Ledger, Q: QuoteBook> DbFetcherThread<DB, Q> {
+    const POLLING_FREQUENCY: Duration = Duration::from_millis(10);
+    const ERROR_RETRY_FREQUENCY: Duration = Duration::from_millis(1000);
+
+    pub fn start(
+        db: DB,
+        quotebook: Q,
+        msg_bus_tx: Sender<Msg>,
+        next_block_index: u64,
+        shared_state: Arc<Mutex<DbPollSharedState>>,
+        stop_requested: Arc<AtomicBool>,
+        logger: Logger,
+    ) {
+        let thread = Self {
+            db,
+            quotebook,
+            msg_bus_tx,
+            next_block_index,
+            shared_state,
+            stop_requested,
+            logger,
+        };
+        thread.run();
+    }
+
+    fn run(mut self) {
+        log::info!(self.logger, "Db fetcher thread started.");
+        self.next_block_index = 0;
```

Comment:
I think, assuming the quote book is a lot smaller than the ledger, there's probably a faster startup procedure here? Where like, you look at current ledger size, ask for each thing in the quote book if it's key image is in the ledger, and once you've done that, you know you've "scanned" up to that block height. and you only start loading individual blocks from that block height on

this can be a follow on PR

Reply:
Thank you. I'll resolve that in a follow up.